### ISSUE TITLE
feat: Grouping in a tree structure - Let's be very organised

### ISSUE THREAD

**Issue by psyray** [Security-Tools-Alliance/rengine-ng#50]
### Is there an existing feature or issue for this?

- [X] I have searched the existing issues

### Expected feature

Proposed by @carboncrystal

At the moment it is quite confusing how Rengine is organized.

**Example:** 
Targets menu do not have grouped assets in a tree structure, currently you have for example target a.com with Project X, target b.com with Project Y, after you have target c.com with project X again and so on. 
This is not so productive, because you cannot see the entire structure of that asset (tree structure).

### Proposed Suggestion
Think how easy it would be if every asset from Target Menu, for example a.com to be listed in a left sidebar and when you click a.com on right side to have everything of that target:
- Recent scan (with date)
- Target Summary
- Subscan History (badge with: not scanned yet > filter by: show scanned / not scanned)
- All subdomains
- All Endpoints
- Scan Findings
- Initiate Scan

Thus the Scan History & Vulnerabilities menu which basically displays global results from all assets can be moved e.g. to the Global Overview, or reordered to be directly listed in the main Dashboard.

This is how the big players organise structure > tree structure: think Burp Pro, for example (in the Burp Target menu you have everything in a tree structure). SOCRADAR does the same, and the examples can go on.
Jira & Confluence (the masters of organization) also organize in a tree structure (Issue Navigator).

My opinion is that we do not have to reinvent the wheel.

SOCRADAR screens

![Screenshot from 2023-12-06 04-53-33](https://github.com/yogeshojha/rengine/assets/106466414/e18a2968-55c8-4ae4-91b1-c56eb72dfb05)
![Screenshot from 2023-12-06 04-53-50](https://github.com/yogeshojha/rengine/assets/106466414/dbed727a-05a1-4415-b2a7-676f60174ec0)
![Screenshot from 2023-12-06 04-56-26](https://github.com/yogeshojha/rengine/assets/106466414/e93e99f7-8e1f-4fff-96f7-e8045080a725)
![Screenshot from 2023-12-06 04-56-55](https://github.com/yogeshojha/rengine/assets/106466414/d2342301-cd8b-4069-8d72-768193cb5420)

**Comment by EmmanuelDevienne** [comment]
Yes, reNgine would benefit from a restructuring of organizations and targets. 
Currently, the approach of creating targets and then assigning them to an organization is backwards. 
It should be the other way around!

The idea of having a tree structure is excellent and should be implemented. However, beyond that, the approach to creating an organization needs significant restructuring.

For example, on the organization's edit page:
Instead of selecting targets, several new elements should be implemented. We should have an **in-scope** form and an **out-of-scope** form. It should also support wildcards.

Example:

- **Organization**
    - **In-Scope**
        - `*.domain1.tld`
        - `domain2.tld`
        - `sub.domain3.tld`
    - **Out-of-Scope**
        - `*.domain4.tld`

Domains without wildcards should not trigger subdomain enumeration.

Additionally, it would be ideal to add global variables for scan engines, such as the **User Agent** and **rate limit**, directly on the organization's edit page.

I believe this approach is optimal. In bug bounty programs, we are often given a defined scope to follow, within which we need to specify user agents for our requests and adhere to a rate limit.

**Comment by sundayz-hunter** [comment]
In addition to what has been said above, and in anticipation of a future reorganization, here's an example of an approach from another framework. 

![image](https://github.com/user-attachments/assets/129017c2-5b7d-4afd-aae8-e3e313fce43b)

It's really necessary to rethink the whole structuring so as to have to fill in a program's information only once, without having to come back to it later. 

A program proposes targets within the scope and others outside the scope, sometimes with a rate limit or particular cookies to be added. 

All this has to be done when creating the program in reNgine and then in the scan yml file you just have to build the tools you want to use according to the type of scan you want, but you don't have to come back to what concerns out-of-scope domains, i.e. whether it's a wildcard domain or not. 

A check must be made to ensure that a domain that is not wildcard does not enter the sub-domain enumeration, even if a scan requests it.
